cartographer_ros/cartographer_ros/map_builder_bridge.cc

```diff
@@ -36,6 +36,22 @@ ::std_msgs::ColorRGBA ToMessage(const cartographer::io::FloatColor& color) {
   return result;
 }
 
+visualization_msgs::Marker CreateTrajectoryMarker(int trajectory_id, int split,
+                                                  const std::string& frame_id) {
+  visualization_msgs::Marker marker;
+  marker.ns = "Trajectory " + std::to_string(trajectory_id) + "." +
+              std::to_string(split);
```

Comment:
Having a separate namespace for each split would lead to many namespaces shown in RViz. Also, the labels will change when the trajectory is split. Any reason why we don't use one namespace per trajectory and increase the `marker.id` only?

Reply:
It was a small indication to show how many splits happened in that trajectory. And I think it can only grow only to num_submaps in the trajectory. I can revert it back if you don't like this. Or I can append node_index instead of num split. How about this?

Reply:
If we remove single nodes there could be more. In what case would you find providing this information in RViz useful? If we keep it one trajectory ID per namespace, hiding everything related to a trajectory ID is much easier which seems a good thing to me.

Reply:
I agree that it will be too many if cartographer removes single nodes. I am fine with having one namespace per trajectory. It was just for convenient debuggability.